packages/fiori/src/DynamicPageTitle.ts

```diff
@@ -0,0 +1,322 @@
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import ResizeHandler from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import type { ResizeObserverCallback } from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import { isEnter, isSpace } from "@ui5/webcomponents-base/dist/Keys.js";
+import type Toolbar from "@ui5/webcomponents/dist/Toolbar.js";
+import type { ToolbarMinWidthChangeEventDetail } from "@ui5/webcomponents/dist/Toolbar.js";
+import ToolbarItemOverflowBehavior from "@ui5/webcomponents/dist/types/ToolbarItemOverflowBehavior.js";
+
+// Template
+import DynamicPageTitleTemplate from "./generated/templates/DynamicPageTitleTemplate.lit.js";
+
+// Styles
+import DynamicPageTitleCss from "./generated/themes/DynamicPageTitle.css.js";
+
+// Texts
+import {
+	DYNAMIC_PAGE_ARIA_DESCR_TOGGLE_HEADER,
+} from "./generated/i18n/i18n-defaults.js";
+
+/**
+ * @class
+ *
+ * <h3>Overview</h3>
+ *
+ * Title of the <code>DynamicPage</code>.
+ *
+ * The <code>DynamicPageTitle</code> component is part of the <code>DynamicPage</code>
+ * family and is used to serve as title of the <code>DynamicPage</code>.
+ *
+ * <h3>Usage</h3>
+ *
+ * The <code>DynamicPageTitle</code> can hold any component and displays the most important
+ * information regarding the object that will always remain visible while scrolling.
+ *
+ * <b>Note:</b> The <code>actions</code> slot accepts any UI5 web component, but it`s
+ * recommended to use <code>ui5-toolbar</code>.
+ *
+ * The user can switch between the expanded/collapsed states of the
+ * <code>DynamicPage</code> by clicking on the <code>DynamicPageTitle</code>
+ * or by using the expand/collapse visual indicators, positioned at the bottom of the
+ * <code>DynamicPageTitle</code> and the <code>DynamicPageHeader</code> inside <code>ui5-dynamic-page-header-actions</code>.
+ *
+ * <h3>Responsive Behavior</h3>
+ *
+ * The responsive behavior of the <code>DynamicPageTitle</code> depends on the behavior of the
+ * content that is displayed.
+ *
+ *
+ * @constructor
+ * @extends UI5Element
+ * @public
+ * @since 1.23.0
+ */
+@customElement({
+	tag: "ui5-dynamic-page-title",
+	fastNavigation: true,
+	renderer: litRender,
+	styles: DynamicPageTitleCss,
+	template: DynamicPageTitleTemplate,
+})
+
+/**
+ * Event is fired when the title is toggled.
+ * @private
+ */
+@event("_toggle-title")
+
+class DynamicPageTitle extends UI5Element {
+	/**
+	 * Defines if the title is snapped.
+	 *
+	 * @public
+	 * @default false
+	 */
+	@property({ type: Boolean })
+	snapped!: boolean;
```

Comment:
Isn't better to make this one private (in the sense of protected/friendly) and only be set by the DynamicPage as it's done DynamicPage#onBeforeRendering, and DynamicPage#snapTitleByScroll methods. Or, the DynamicPageTitle is also meant to be used outside the DynamicPage.